### PR TITLE
refactor(permissions): restricted-by-permissions to rtl

### PR DESCRIPTION
**Pull request by tdeekens**
#### Summary

This pull request refactors, as promised, the `RestrictedByPermissions` to use RTL.

### CODE REVIEW

File: packages/permissions/src/components/restricted-by-permissions/restricted-by-permissions.spec.tsx [hunk 1]
```diff
@@ -20,142 +18,222 @@ type TDemandedActionRight = {
...
 };
-type TApplicationContext = {
-  permissions: TPermissions | null;
-  actionRights: TActionRights | null;
-};
-type TestProps = {
-  shouldMatchSomePermissions: boolean;
-  permissions: TPermissionName[];
+
+const testRender = ({
```
Comment (tdeekens) [inline packages/permissions/src/components/restricted-by-permissions/restricted-by-permissions.spec.tsx]: Setup here is similar to our tests in `use-is-authorized`.

File: packages/permissions/src/components/restricted-by-permissions/restricted-by-permissions.spec.tsx [hunk 2]
```diff
@@ -20,142 +18,222 @@ type TDemandedActionRight = {
...
+        cdnUrl: 'http://localhost:3001',
+        servedByProxy: false,
+      }}
+      projectDataLocale="en"
+    >
+      {component}
+    </ApplicationContextProvider>
+  );
+
+const FaaCTestComponent = (props: {
```
Comment (tdeekens) [inline packages/permissions/src/components/restricted-by-permissions/restricted-by-permissions.spec.tsx]: We have a simple component per "style" we want to test.

File: packages/permissions/src/components/restricted-by-permissions/restricted-by-permissions.spec.tsx [hunk 3]
```diff
@@ -20,142 +18,222 @@ type TDemandedActionRight = {
...
-  permissions: {
-    canManageProjectSettings: true,
-  },
-  actionRights: {},
-  ...custom,
-});
+describe('with permissions', () => {
+  describe('when demanded permissions are present', () => {
+    describe('with `FaaC`', () => {
+      it('should indicatged being authorzied', () => {
```
Comment (emmenko) [inline packages/permissions/src/components/restricted-by-permissions/restricted-by-permissions.spec.tsx]:
```suggestion
      it('should indicate being authorized', () => {
```

File: packages/permissions/src/components/restricted-by-permissions/restricted-by-permissions.spec.tsx [hunk 4]
```diff
@@ -20,142 +18,222 @@ type TDemandedActionRight = {
...
-    beforeEach(() => {
-      props = createTestProps({ children: jest.fn() });
-      shallow(<RestrictedByPermissions {...props} />)
-        .find(ApplicationContext)
-        .renderProp('render')(createApplicationContext())
-        .find(Authorized)
-        .renderProp('render')(true);
+
+    describe('with `render`-prop', () => {
+      it('should indicatged being authorzied', () => {
```
Comment (emmenko) [inline packages/permissions/src/components/restricted-by-permissions/restricted-by-permissions.spec.tsx]:
```suggestion
      it('should indicate being authorized', () => {
```

File: packages/permissions/src/components/restricted-by-permissions/restricted-by-permissions.spec.tsx [hunk 5]
```diff
@@ -20,142 +18,222 @@ type TDemandedActionRight = {
...
+        });
+
+        expect(rendered.queryByText('Is authorized: Yes')).toBeInTheDocument();
+      });
     });
-    it('should call children prop with isAuthorized true', () => {
-      expect(props.children).toHaveBeenCalledWith({ isAuthorized: true });
+
+    describe('with `children` being a `ReactNode`', () => {
+      it('should indicatged being authorzied', () => {
```
Comment (emmenko) [inline packages/permissions/src/components/restricted-by-permissions/restricted-by-permissions.spec.tsx]:
```suggestion
      it('should indicate being authorized', () => {
```

File: packages/permissions/src/components/restricted-by-permissions/restricted-by-permissions.spec.tsx [hunk 6]
```diff
@@ -20,142 +18,222 @@ type TDemandedActionRight = {
...
-          </RestrictedByPermissions>
-        )
-          .find(ApplicationContext)
-          .renderProp('render')(createApplicationContext())
-          .find(Authorized)
-          .renderProp('render')(true);
+
+  describe('when demanded permissions are not present', () => {
+    describe('with `FaaC`', () => {
+      it('should indicatged not being authorzied', () => {
```
Comment (emmenko) [inline packages/permissions/src/components/restricted-by-permissions/restricted-by-permissions.spec.tsx]:
```suggestion
      it('should indicate not being authorized', () => {
```

File: packages/permissions/src/components/restricted-by-permissions/restricted-by-permissions.spec.tsx [hunk 7]
```diff
@@ -20,142 +18,222 @@ type TDemandedActionRight = {
...
+          component: <FaaCTestComponent />,
+        });
+
+        expect(rendered.queryByText('Is authorized: No')).toBeInTheDocument();
       });
-      it('should render children', () => {
-        expect(wrapper).toHaveText('Authorized');
+    });
+    describe('with `render`-prop', () => {
+      it('should indicatged not being authorzied', () => {
```
Comment (emmenko) [inline packages/permissions/src/components/restricted-by-permissions/restricted-by-permissions.spec.tsx]:
```suggestion
      it('should indicate not being authorized', () => {
```

File: packages/permissions/src/components/restricted-by-permissions/restricted-by-permissions.spec.tsx [hunk 8]
```diff
@@ -20,142 +18,222 @@ type TDemandedActionRight = {
...
-          <RestrictedByPermissions {...props}>
-            <div>{'Authorized'}</div>
-          </RestrictedByPermissions>
-        )
-          .find(ApplicationContext)
-          .renderProp('render')(createApplicationContext())
-          .find(Authorized)
-          .renderProp('render')(false);
+    describe('with `children` being a `ReactNode`', () => {
+      it('should indicatged not being authorzied', () => {
```
Comment (emmenko) [inline packages/permissions/src/components/restricted-by-permissions/restricted-by-permissions.spec.tsx]:
```suggestion
      it('should indicate not being authorized', () => {
```

File: packages/permissions/src/components/restricted-by-permissions/restricted-by-permissions.spec.tsx [hunk 9]
```diff
@@ -20,142 +18,222 @@ type TDemandedActionRight = {
...
+
+        expect(
+          rendered.queryByText('Is authorized: Yes')
+        ).not.toBeInTheDocument();
       });
-      it('should render null', () => {
-        expect(wrapper.type()).toBe(null);
+    });
+    describe('with `unauthorizedComponent` being a `ReactNode`', () => {
+      it('should indicatged not being authorzied', () => {
```
Comment (emmenko) [inline packages/permissions/src/components/restricted-by-permissions/restricted-by-permissions.spec.tsx]:
```suggestion
      it('should indicate not being authorized', () => {
```

File: packages/permissions/src/components/restricted-by-permissions/restricted-by-permissions.spec.tsx [hunk 10]
```diff
@@ -20,142 +18,222 @@ type TDemandedActionRight = {
...
-            .renderProp('render')(createApplicationContext())
-            .find(Authorized)
-            .renderProp('render')(false);
+  });
+});
+
+describe('with action rights', () => {
+  describe('when demanded action rights are present', () => {
+    describe('with `FaaC`', () => {
+      it('should indicatged being authorzied', () => {
```
Comment (emmenko) [inline packages/permissions/src/components/restricted-by-permissions/restricted-by-permissions.spec.tsx]:
```suggestion
      it('should indicate not being authorized', () => {
```

File: packages/permissions/src/components/restricted-by-permissions/restricted-by-permissions.spec.tsx [hunk 11]
```diff
@@ -20,142 +18,222 @@ type TDemandedActionRight = {
...
-        it('should render unauthorizedComponent', () => {
-          expect(wrapper).toRender(Unauthorized);
+
+        expect(rendered.queryByText('Is authorized: Yes')).toBeInTheDocument();
+      });
+    });
+  });
+  describe('when demanded action rights are not present', () => {
+    describe('with `FaaC`', () => {
+      it('should indicatged being not authorzied', () => {
```
Comment (emmenko) [inline packages/permissions/src/components/restricted-by-permissions/restricted-by-permissions.spec.tsx]:
```suggestion
      it('should indicate not being authorized', () => {
```

**Pull request merged**